src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java

```diff
@@ -46,7 +47,7 @@
  * @author o_sukhosolsky
  */
 public abstract class AbstractHeaderCheck extends AbstractFileSetCheck
-    implements ExternalResourceHolder {
+    implements ExternalResourceHolder, OneModuleInstancePerApplication {
```

Comment:
> Also why didn't you mark AbstractCheck and AbstractFileSetCheck but AbstractJavadocCheck. Once you mark the base, all checks are now forced under that if it is the strongest type.

As far as I remember, there was a problem with `blockCommentAst` field. I think, it is better to move `AbstractJavadocCheck` changes to another PR